src/Discord.Net/Entities/Messages/ITag.cs

```diff
@@ -0,0 +1,14 @@
+// todo: docs
+using System;
+
+namespace Discord
+{
+    public interface ITag
+    {
+        Index Index { get; } // review: is this big brained? would it be bigger brained to just use a ReadOnlySpan?
```

Comment:
Spans aren't really user-friendly (imo) so for user-experience probably better to leave as an Index.

Reply:
What purpose can a Span even serve to replace an Index?